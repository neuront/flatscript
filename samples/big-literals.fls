x: 1000000000000 * 2 % 31
console.log(x)
console.log(1000000000000 * 2 % 37)
y: 8.0 * 1000000000000 / 2500000000000
console.log(y)

console.log((1 << 30) + (1 << 30))
console.log(((1 << 30) + (1 << 30)) << 1)
console.log(-1 >> 1)
console.log(-1 >>> 1)

z: 1 + 2
console.log(z.toString())
