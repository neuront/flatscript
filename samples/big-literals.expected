8
2
3.2
2147483648
0
-1
2147483647
3
